gnovm/cmd/gno/fmt.go

```diff
@@ -0,0 +1,307 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"go/format"
+	"go/parser"
+	"go/scanner"
+	"go/token"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/gnolang/gno/gnovm/pkg/gnoenv"
+	"github.com/gnolang/gno/gnovm/pkg/gnoimports"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"github.com/rogpeppe/go-internal/diff"
+)
+
+type fmtCfg struct {
+	write   bool
+	quiet   bool
+	diff    bool
+	verbose bool
+	imports bool
+	strict  bool
+	include fmtIncludes
+}
+
+var defaultFmtOptions = &fmtCfg{
+	imports: true,
+}
+
+func newFmtCmd(io commands.IO) *commands.Command {
+	cfg := &fmtCfg{}
+	return commands.NewCommand(
+		commands.Metadata{
+			Name:       "fmt",
+			ShortUsage: "gno fmt [flags] [path ...]",
+			ShortHelp:  "Run gno file formatter.",
+			LongHelp:   "The `gno fmt` tool processes, formats, and cleans up `gno` source files.",
+		},
+		cfg,
+		func(_ context.Context, args []string) error {
+			return execFmt(cfg, args, io)
+		})
+}
+
+func (c *fmtCfg) RegisterFlags(fs *flag.FlagSet) {
+	fs.BoolVar(
+		&c.write,
+		"w",
```

Comment:
Hmm, I think we want to be consistent with most `go fmt` tools and keep at least the `-w` flag, as it will come naturally to most people. Other flags like `i` and `q` could become long flags, which I think is acceptable.